lib/puppet/provider.rb

```diff
@@ -249,12 +251,20 @@ def environment(env)
       @custom_environment = @custom_environment.merge(env)
     end
 
+    def resolver(res)
+      @command_resolver = res
+    end
+
+    def executor(exec)
+      @command_executor = exec
+    end
+
     def command
       if not @optional
         @confiner.confine :exists => @path, :for_binary => true
       end
 
-      Puppet::Provider::Command.new(@name, @path, Puppet::Util, Puppet::Util::Execution, { :failonfail => true, :combine => true, :custom_environment => @custom_environment })
+      Puppet::Provider::Command.new(@name, @path, @command_resolver, @command_executor, { :failonfail => true, :combine => true, :custom_environment => @custom_environment })
```

Comment:
Similarly, I think you can just pass `self` here, e.g. `Command.new(@name, @path, self, ...)`